src/arbitrary/array.ts

```diff
@@ -34,19 +34,34 @@ export interface ArrayConstraints {
    * @remarks Since 2.22.0
    */
   size?: SizeForArbitrary;
+  /**
+   * When receiving a depth identifier, the arbitrary will impact the depthFactor
+   * attached to it to avoid going too deep if it already generated lots of items.
+   *
+   * In other words, if the number of generated values within the collection is large
+   * then the generated items will tend to be less deep to avoid creating structures a lot
+   * larger than expected.
+   *
+   * For the moment, the depth is not taken into account to compute the number of items to
+   * define for a precise generate call of the array. Just applied onto eligible items.
+   *
+   * @remarks Since 2.25.0
+   */
+  depthIdentifier?: string;
```

Comment:
In another PR, we should factorize all the "depthIdentifier" so that we use a single and centralized type for them. We should also offer a creator for such identifier to stop relying on strings reprensenting what's suppose to be a unique identifier.